src/components/SearchBar/SearchBarDesktop.tsx

```diff
@@ -0,0 +1,274 @@
+import MenuItemContainer from './MenuItemContainer';
+import useDimensions from './use-dimensions';
+
+type SearchBarDesktopProps = {
+  className: string;
+  activityPictures: Array<{
+    thumbnail: string;
+    url: string;
+    description: string;
+  }>;
+  activityKeywords: Array<{
+    url: string;
+    keyword: string;
+  }>;
+  locations: Array<{
+    url: string;
+    text: string;
+  }>;
+  categories: Array<{
+    icon: LucideIcon;
+    url: string;
+    text: string;
+  }>;
+};
+
+const locationMenu = {
+  open: (height = 1000) => ({
+    clipPath: `circle(${height * 2 + 200}px at 40px 800px)`,
+    transition: {
+      type: 'spring',
+      stiffness: 7,
+      restDelta: 2,
+    },
+  }),
+  closed: {
+    // TODO: use mouse event to get cursor position
+    clipPath: 'circle(0px at 40px 800px)',
+    transition: {
+      delay: 0.5,
+      type: 'spring',
+      stiffness: 400,
+      damping: 40,
+    },
+  },
+};
+
+const categoryMenu = {
+  open: (height = 1000) => ({
+    clipPath: `circle(${height * 2 + 200}px at 340px 800px)`,
+    transition: {
+      type: 'spring',
+      stiffness: 7,
+      restDelta: 2,
+    },
+  }),
+  closed: {
+    // TODO: use mouse event to get cursor position
+    clipPath: 'circle(0px at 340px 800px)',
+    transition: {
+      delay: 0.5,
+      type: 'spring',
+      stiffness: 400,
+      damping: 40,
+    },
+  },
+};
+
+const SearchBarDesktop = ({
+  className = '',
+  activityPictures,
+  activityKeywords,
+  locations,
+  categories,
+}: SearchBarDesktopProps) => {
+  const [isSearchBarMenuOpen, setSearchBarMenuOpen] = useState(false);
+  const [isCategoryMenuOpen, setCategoryMenuOpen] = useState(false);
+  const [isLocationMenuOpen, setLocationMenuOpen] = useState(false);
+  const containerRef = useRef(null);
+  const { height } = useDimensions(containerRef);
+
+  return (
+    <section
+      className={cn(
+        'sticky top-72 z-20 mx-auto max-w-[81rem] space-y-6 border-t border-primary bg-surface py-6 text-primary',
+        className
+      )}
+    >
+      <H3>依照需求搜尋適合你的活動</H3>
+      <form
+        className="flex gap-2"
+        action={(formData) => {
+          // TODO: server action for search data
+          console.log(formData);
+        }}
+      >
+        <div className="flex grow">
+          <Popover open={isSearchBarMenuOpen}>
+            <PopoverTrigger />
+            <PopoverContent
+              sticky="always"
+              side="top"
+              align="start"
+              sideOffset={0}
+              className="mx-auto min-w-[81rem] border-x border-t border-primary bg-surface py-12"
+              onPointerDownOutside={() => setSearchBarMenuOpen(() => false)}
+              onEscapeKeyDown={() => setSearchBarMenuOpen(() => false)}
+              // prevent auto focusing for input text
+              onOpenAutoFocus={(e) => e.preventDefault()}
+            >
+              <p className="ml-3 text-base font-bold">推薦活動</p>
```

Comment:
Popover 左側間距為 `16px`
```suggestion
              <p className="ml-4 text-base font-bold">推薦活動</p>
```